File: src/models/security-modal.ts
import type { DoubleAuthMode } from "./double-auth-mode";
import type { PasswordRules } from "./password-rules";

export type SecurityModal = Readonly<{
  availableSecurityModes: DoubleAuthMode[];
  defaultSecurityMode: DoubleAuthMode;
  passwordRules: PasswordRules;

  shouldCustomPassword: boolean;
  shouldCustomDoubleAuth: boolean;

  /**
   * Should be internal use only.
   */
  context: {
    authentication: any,
    identity: any,
    initialUsername?: string
  }
}>;

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/SecurityModal.kt
package ink.literate.pawnote.models

import ink.literate.pawnote.api.private.IdentifyResponse
import kotlinx.serialization.json.JsonObject

data class SecurityModalContext(
    var authentication: JsonObject,
    val identity: IdentifyResponse,
    val initialUsername: String? = null
)

data class SecurityModal(
    val availableSecurityModes: List<DoubleAuthMode>,
    val defaultSecurityMode: DoubleAuthMode,
    val passwordRules: PasswordRules,
    val shouldCustomPassword: Boolean,
    val shouldCustomDoubleAuth: Boolean,
    val shouldEnterPIN: Boolean,
    val shouldEnterSource: Boolean,

    /** Should be internal use only. */
    val context: SecurityModalContext
)
